trigger.tex

```diff
@@ -39,9 +40,84 @@ \chapter{Trigger Module}
     bits of \RcsrTselect but fewer than $2^m$ triggers.
 \end{commentary}
 
-It is possible for a trigger with the ``enter Debug Mode'' action (1) and another
-trigger with the ``raise a breakpoint exception'' action (0) to fire at the same
-time. The preferred behavior is to have both actions take place.  It is
+\section{Actions}
+
+Triggers can be configured to take one of several actions when they fire.
+Table~\ref{tab:action} lists all options.
+
+\begin{table}[H]
+\centering
+\caption{\FcsrMcontrolAction encoding}
+\label{tab:action}
+\begin{tabular}{|r|L|}
+\hline
+Value & Description \\
+\hline
+0 & Raise a breakpoint exception into M-Mode. (Used when software wants to
+    use the trigger module without an external debugger attached.)
+    If the trigger is precise, \Rmepc must contain the address of the
+    instruction that caused the trigger to fire. If the trigger is imprecise
+    \Rmepc will contain the address of the next instruction to be executed. \\
+\hline
+1 & Enter Debug Mode. (Only supported when the trigger's \FcsrTdataOneDmode
```

Comment:
What does "supported" mean?  If someone writes 1 here but tdata1.dmode=0, what happens?  We could say that this bit is WARL and the only legal value when dmode=0 is 0.  But that doesn't work if debug mode writes tdata1.dmode=1 (perfectly legal), we write this bit to 1 (perfectly legal at this time so no WARL issues), then debug mode writes tdata1.dmode=0 (also legal).  Now we have inconsistent/illegal state.

Should there just be a note here that says that if tdata1.dmode=0 then this bit is treated as 0 regardless of the actual value?

Reply:
I agree these bits (and a bunch of others) should be WARL.

You can't write individual bits in a register so I don't think your example can happen. There are instructions that purport to set/clear individual CSR bits, but I always think of them as being short-hand for reading the whole register, flipping a bit, and writing the whole thing. Am I wrong about that?

Reply:
I believe that you're right about the read-modify-write CSR instructions.  Let me consider my corner case more carefully and see if I still think it's a problem.

Reply:
I was confused.  They're presented in different tables so I didn't see that dmode and action were in the same register.

I agree that this isn't an issue as long as this field is WARL and it's illegal to have dmode=0 and action=1.  I think that the word "supported" should be changed to "legal" to make it clear that this case will be handled by the normal WARL mechanisms rather than some other mechanism that ignores state that's not "supported."

It's not the most pressing issue but incremental improvements are still useful.

```suggestion
1 & Enter Debug Mode. (Only legal when the trigger's \FcsrTdataOneDmode
```